File: gateway-server/src/main/java/org/lucycato/gatewayserver/application/port/in/AuthGlobalFilterUseCase.java
package org.lucycato.gatewayserver.application.port.in;

import reactor.core.publisher.Mono;

public interface AuthGlobalFilterUseCase {
    Mono<String> parseJwtToAdminUserJson(String token);

    Mono<String> parseJwtToAppUserJson(String token);
}

File: gateway-server/src/main/kotlin/org/lucycato/gatewayservice/application/port/in/AuthGlobalFilterUseCase.kt
package org.lucycato.gatewayservice.application.port.`in`

import reactor.core.publisher.Mono

interface AuthGlobalFilterUseCase {
    fun parseJwtToAdminUserJson(token: String?): Mono<String>

    fun parseJwtToAppUserJson(token: String?): Mono<String>
}